src/webview/flink-statement-results.ts

```diff
@@ -285,6 +280,18 @@ export class FlinkStatementResultsViewModel extends ViewModel {
     }, null);
   }
 
+  async setViewMode(viewMode: ViewMode) {
+    if (viewMode === "table") {
+      this.changelogPage(this.page());
+      this.page(this.tablePage());
+    } else {
+      this.tablePage(this.page());
+      this.page(this.changelogPage());
+    }
+
+    await this.post("SetViewMode", { viewMode, timestamp: this.timestamp() });
+  }
```

Comment:
This makes it so that the page being viewed in each of the changelog/table views is preserved when switching back and forth, hence offering a nice experience.